### PR TITLE
[BIP 2] License header in preamble

### PR DESCRIPTION
**Pull request by MarcoFalke**

### CODE REVIEW

**Comment by luke-jr** [comment]
Would be nice to use an existing abbreviation list. Maybe [Gentoo's](https://gitweb.gentoo.org/repo/gentoo.git/tree/licenses)?

Also, it's not clear how one might specify needing two different licenses, such as when example code is not under the same licenses as the BIP text.

(and I'd prefer to disallow PD in new BIPs...)

**Comment by harding** [comment]
@luke-jr do you have a write-up or link to why you'd like to disallow PD in future BIPs?  Using PD is my  preference for my own self-published documentation, but I'm interested in learning why you think it's a bad idea.

**Comment by luke-jr** [comment]
@harding Releasing into PD is not a legally valid action in some jurisdictions.

**Comment by harding** [comment]
@luke-jr ah, fair point.  How do you feel about near alternatives like [CC0](https://creativecommons.org/publicdomain/zero/1.0/legalcode) that put the doc in PD if possible but have a unlimited-use copyright fallback when that isn't an option?

**Comment by luke-jr** [comment]
@harding CC0 is already on the list...

**Comment by harding** [comment]
@luke-jr ah, sorry.  I overlooked that.  Anyway, concept ACK on not using PD for future BIPs as long as CC0 or similar public-domain-or-backup options are available.

**Comment by luke-jr** [comment]
@MarcoFalke ping?

**Comment by MarcoFalke** [comment]
Sorry, was busy last week. Will try to get back to this within some days.

**Comment by MarcoFalke** [comment]
> Would be nice to use an existing abbreviation list. Maybe Gentoo's?

I could not find GNU-ALL-PERMISSIVE so I will just leave it as is.

> Also, it's not clear how one might specify needing two different licenses, such as when example code is not under the same licenses as the BIP text.

Added a commit to address this concern.

> (and I'd prefer to disallow PD in new BIPs...)

It is already in the "not recommended" section but you can't forbid someone using it. Also we need the abbreviation to apply it to the other bips (grandfather policy).

**Comment by luke-jr** [comment]
Can't we forbid PD moving forward? It's essentially the same as no license at all in some places... (but yes, I agree we need an abbreviation for grandfathered-in stuff.)

File: bip-0002.mediawiki [hunk 1]
```diff
@@ -133,31 +133,46 @@ Will BIP comments be censored or limited to particular participants/"experts"?
 
 ==BIP licensing==
 
-New BIPs may be accepted with the following licenses:
-
 ===Specification===
 
+New BIPs may be accepted with the following licenses. Each new BIP must identify at least one acceptable license in its preamble. The License header in the preamble must be placed after the Created header. Each license must be referenced by their respecitve abbreviation given below.
```
Comment (luke-jr) [inline bip-0002.mediawiki]: I don't think we should start defining ordering of headers here.

Also, it is undefined how to interpret multiple licenses (OR vs AND; probably want OR).

**Comment by luke-jr** [comment]
Where are the abbreviations from?

**Comment by MarcoFalke** [comment]
> Where are the abbreviations from?


I have tried to stick to what their respective link says, or what appears to be commonly used. You are welcome to replace them with the Gentoo abbreviations but I don't think this is important as long as it is understandable/not misleading and unique.

File: bip-0002.mediawiki (hunk 1 above)
Comment (MarcoFalke) [inline bip-0002.mediawiki]: There is a strict order defined in https://github.com/bitcoin/bips/blob/master/bip-0001.mediawiki#bip-header-preamble so let's jut keep a strict order.

**Pull request merged**